src/if_checks.c

```diff
@@ -400,7 +415,12 @@ struct check_result *check_name_required_but_not_used_in_if(const struct
 		return NULL;
 	}
 
-	struct name_list *names_to_check = get_names_in_node(node);
+	struct name_list *names_to_check;
+	if (dd->flavor == DECL_CLASS) {
```

Comment:
Why do we need special handling here instead of just teaching `get_names_in_node()` about classes?

Reply:
Turns out `get_names_in_node()` already supported that case; dropped the special handling here.